sechub-wrapper-xray/src/test/java/com/mercedesbenz/sechub/wrapper/xray/cli/XrayWrapperArtifactoryClientSupportTest.java

```diff
@@ -0,0 +1,53 @@
+package com.mercedesbenz.sechub.wrapper.xray.cli;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedConstruction;
+import org.mockito.Mockito;
+
+import com.mercedesbenz.sechub.wrapper.xray.XrayWrapperException;
+import com.mercedesbenz.sechub.wrapper.xray.api.XrayAPIArtifactoryClient;
+import com.mercedesbenz.sechub.wrapper.xray.config.XrayWrapperArtifact;
+import com.mercedesbenz.sechub.wrapper.xray.config.XrayWrapperConfiguration;
+
+class XrayWrapperArtifactoryClientSupportTest {
+
+    XrayWrapperArtifactoryClientSupport clientSupportToTest;
+
+    XrayWrapperConfiguration configuration;
+
+    XrayWrapperArtifact artifact;
+
+    @BeforeEach
+    void beforeEach() {
+        configuration = mock(XrayWrapperConfiguration.class);
+        artifact = mock(XrayWrapperArtifact.class);
+    }
+
+    @Test
+    void waitForScansToFinishAndDownloadReport_valid_execution_without_report_transformation() throws XrayWrapperException {
+        /* test + execute */
+        try (MockedConstruction<XrayAPIArtifactoryClient> mockedClient = Mockito.mockConstruction(XrayAPIArtifactoryClient.class, (mock, context) -> {
+            when(mock.requestXrayVersion()).thenReturn("mocked-version");
+            when(mock.checkArtifactoryUploadSuccess()).thenReturn(true);
+            when(mock.getScanStatus()).thenReturn(XrayWrapperArtifactoryClientSupport.ScanStatus.SCANNED);
+            when(mock.requestScanReports()).thenReturn(false);
+        })) {
+            clientSupportToTest = new XrayWrapperArtifactoryClientSupport(configuration, artifact);
+            clientSupportToTest.waitForScansToFinishAndDownloadReport();
+        }
+    }
+
+    @Test
+    void waitForScansToFinishAndDownloadReport_throws_XrayWrapperException() {
```

Comment:
I do not exactly get the point why the exception is thrown.

looking at the test before there are 4 mock calls/definitions to get it working - what is here missing to bring up the exception? Or do we have 4 situation were the exception could happen? In this case we should have 4 test methods where only one of the parameters is missing to test the combinations.